src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+        case 0:
+            obj2F4->unk_18C[0].x = D_800C9FA0[obj2F4->index].x + gPlayer->pos.x;
+            obj2F4->unk_18C[0].y = D_800C9FA0[obj2F4->index].y + gPlayer->pos.y;
+            obj2F4->unk_18C[0].z = D_800C9FA0[obj2F4->index].z + gPlayer->pos.z;
+            Math_SmoothStepToF(&obj2F4->obj.pos.x, obj2F4->unk_18C[0].x, 0.05f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&obj2F4->obj.pos.y, obj2F4->unk_18C[0].y, 0.05f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&obj2F4->obj.pos.z, obj2F4->unk_18C[0].z, 0.05f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&obj2F4->unk_0F4.x, gPlayer->unk_0E4, 0.1f, 2.0f, 0.0001f);
+            Math_SmoothStepToF(&obj2F4->unk_0F4.y, gPlayer->unk_0E8, 0.1f, 2.0f, 0.0001f);
+            Math_SmoothStepToF(&obj2F4->unk_0F4.z, 0.0f, 0.05f, 0.2f, 0.0001f);
+            break;
+
+        case 1:
+            obj2F4->unk_0B8 = 2;
+            Audio_PlaySfx(0x09000002U, &obj2F4->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            obj2F4->timer_0BC = 0x96;
+            obj2F4->unk_188 = 5.0f;
+
+        case 2:
+            obj2F4->unk_07C = 2;
+            obj2F4->unk_114 += 2.0f;
+            if (obj2F4->timer_0BC == 0) {
+                Object_Kill(&obj2F4->obj, &obj2F4->sfxPos);
+            }
+            break;
+    }
+    Matrix_RotateY(gCalcMatrix, (obj2F4->unk_0F4.y + 180.0f) * M_DTOR, 0);
+    Matrix_RotateX(gCalcMatrix, -(obj2F4->unk_0F4.x * M_DTOR), 1);
+    sp3C.x = 0.0f;
+    sp3C.y = 0.0f;
+    sp3C.z = obj2F4->unk_114;
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp3C, &sp30);
+    obj2F4->unk_0E8.x = sp30.x;
+    obj2F4->unk_0E8.y = sp30.y;
+    obj2F4->unk_0E8.z = sp30.z;
+    obj2F4->obj.rot.x = -obj2F4->unk_0F4.x;
+    obj2F4->obj.rot.y = obj2F4->unk_0F4.y + 180.0f;
+    obj2F4->obj.rot.z = -obj2F4->unk_0F4.z;
+}
+
+void func_8004990C(Player* player) {
+    if (D_801784AC == 0) {
+        D_80177CE8 += 60.0f;
+    }
+    player->unk_138 = player->pos.z + player->unk_08C;
+    player->unk_0F8 = player->unk_0EC + player->unk_12C + player->unk_130;
+}
+
+void func_80049968(Object_2F4* arg0, s32 arg1) {
+    Object_2F4_Initialize(arg0);
+    arg0->obj.status = 1;
+    arg0->obj.id = 0xC3;
+    arg0->obj.pos.x = D_800C9FD0[arg1];
+    arg0->obj.pos.y = D_800C9FE0[arg1];
+    arg0->obj.pos.z = D_800C9FF0[arg1] + (((void) 0, gPlayer))->unk_138;
+    arg0->unk_114 = D_800CA000[arg1];
+    arg0->unk_118 = D_800CA010[arg1];
+    arg0->unk_0B6 = D_800CA030[arg1];
+    arg0->unk_0B8 = D_800CA040[arg1];
+    arg0->obj.rot.z = D_800CA020[arg1];
+    arg0->obj.rot.y = 180.0f;
+    arg0->unk_0E8.z = gPlayer->vel.z;
+    Object_SetInfo(&arg0->info, arg0->obj.id);
+    arg0->unk_07C = 1;
+    Audio_PlaySfx(0x3100000CU, &arg0->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_80049A9C(Object_8C* obj8c, f32 x, f32 y, f32 z) {
+    Object_8C_Initialize(obj8c);
+    obj8c->obj.status = 1;
+    obj8c->obj.id = 0x15A;
+    obj8c->timer_50 = 0x64;
```

Comment:
```suggestion
    obj8c->timer_50 = 100;
```
timer values should be decimal too.